core/utils/contracts/toStatusName.ts

```diff
@@ -0,0 +1,49 @@
+import type {
+  ExtrinsicStatus,
+  SubmittableResult,
+} from "@polkadot/types/interfaces";
+
+type I18n = { [key: string]: string };
+
+const defaultI18n: I18n = {
+  None: "None",
+  DryRun: "Dry Run",
+  PendingSignature: "Pending signature",
+  Future: "Future",
+  Ready: "Ready",
+  Broadcast: "Broadcast",
+  InBlock: "In block",
+  Retracted: "Retracted",
+  FinalityTimeout: "Finality timeout",
+  Finalized: "Finalized",
+  Usurped: "Usurped",
+  Dropped: "Dropped",
+  Invalid: "Invalid",
+  Valid: "Valid",
+  Dispatched: "Dispatched",
+  DispatchedByBlock: "Dispatched by block",
+  Archived: "Archived",
+  Failed: "Failed",
+  Aborted: "Aborted",
+  TransactionPayment: "Transaction payment",
+  FuturePayment: "Future payment",
+  FinalizedPayment: "Finalized payment",
+};
+
+export const toStatusName = (
+  status: ExtrinsicStatus | SubmittableResult | undefined,
+  i18n: I18n = defaultI18n
```

Comment:
This is a good exploration. I created that issue because this is a common task that could be made easier, or we could come up with better patterns to DRY up code.

1. We want to be able to use [string interpolation](https://www.i18next.com/translation-function/interpolation) too. 
2. We also do not want to introduce a brand new system for handling translations. [i18n-next](https://www.i18next.com/), `react-intl`, etc already exist. Is there a way we can make them easier to use with `useink`? ( I don't know the answer to that yet).

Reply:
Here is a Dapp using translations. I don't think there are examples of interpolation in there, but maybe those libraries are enough and we should not worry about it for `useink`. I'm curious to hear your thoughts